content/docs/user-guide/setup-dropbox-remote.md

```diff
@@ -0,0 +1,82 @@
+# Setup a Dropbox DVC Remote
+
+In this guide we explain the existing ways to setup Dropbox
+[remote storage](/doc/command-reference/remote) for your <abbr>DVC
+projects</abbr>, along with the different benefits each one brings.
+
+## Quick start
+
+To start using a Google Drive remote, you only need to add it with a
+[valid URL format](#url-format). Then use any DVC command that needs to connect
+to it (e.g. `dvc pull` or `dvc push` once there's tracked data to synchronize).
+For example:
+
+```dvc
+$ dvc add data
+...
+$ dvc remote add --default myremote \
+                           gdrive://0AIac4JZqHhKmUk9PDA/dvcstore
+$ dvc push
+
+Go to the following link in your browser:
+
+    https://www.dropbox.com/oauth2/authorize?response_type=code&client_id=<client-id>&token_access_type=offline # ... copy this link
+
+Enter verification code: # <- enter resulting code
+```
+
+See [Authorization](#authorization) for more details.
+
+## URL format
+
+You can indicate any directory name that will be created under the
+`DROPBOX_HOME/Applications/dvc`, e.g.
```

Comment:
in this case (since we rely on DVC to cache data) would it be better to set this directory to avoid sync with local? or if that is not possible use this as a DVC project cache? Though it means that we have to have enough space locally for the whole remote, all remote objects?

btw, if we have enough space locally then we don't even need a special remote- we can create `DROPBOX_HOME/Applications/dvc/my-project` and set it as cache with `dvc cache dir`?

Reply:
It completely misses the point:

- Dropbox uses smart sync. With using Dropbox API you get no network traffic on development machine when the remote experimentation server pushes changes to remote cache.
- You don't actually install Dropbox on a remote machine, especially that a lot of experimentation nowadays happen in a remote Kubernetes cluster (even when on-prem).

Reply:
> You don't actually install Dropbox on a remote experimentation machine

good point, we don't expect this to be installed. And it makes sense that usually you don't want all the data from the remote on any machine.

> Dropbox uses smart sync.

that's what I was referring to. Do we need to specify via API somehow that the dir (in this case `Applications/dvc/something`) should not be synced, or will it do this automatically?

Reply:
It looks like it's not possible to share it https://www.dropboxforum.com/t5/Dropbox-API-Support-Feedback/Sharing-of-folders-inside-the-Apps-folder/td-p/38534 . It makes "App folder" apps limited, right?

Also, another reason to make `Scoped App` vs `App folder` is it's ability to do external deps/outs/imports- (e.g. `dvc import-url dropbox://something` to get and track the dir, `dvc -d dropbox://something -o dropbox://model.pkl`, etc). It is considered advanced indeed, and is rarely used. But we need to test apps that are Scoped, and default dvc app probably should be Scoped.

Reply:
> that's what I was referring to. Do we need to specify via API somehow that the dir (in this case Applications/dvc/something) should not be synced, or will it do this automatically?

It's enabled by default in Drobpox, but a user can adjust that.

> It looks like it's not possible to share it https://www.dropboxforum.com/t5/Dropbox-API-Support-Feedback/Sharing-of-folders-inside-the-Apps-folder/td-p/38534 . It makes "App folder" apps limited, right?

You're right, great catch. Anyway, that would probably influence only the Dropbox App definition, not the integration code. One would rather need to specify a certain directory outside of the `Applications/dvc`. I'll test that.

> Also, another reason to make Scoped App vs App folder is it's ability to do external deps/outs/imports- (e.g. dvc import-url dropbox://something to get and track the dir, dvc -d dropbox://something -o dropbox://model.pkl, etc). It is considered advanced indeed, and is rarely used. But we need to test apps that are Scoped, and default dvc app probably should be Scoped.

Honestly, I didn't even know one could do `dvc -d dropbox://something -o dropbox://model.pkl`. What changes would be required to make it work?

Reply:
> It's enabled by default in Drobpox, but a user can adjust that.

Sorry, I haven't tried it. So, does it mean that if I have an app installed and have a pro account it won't be syncing any new directories, or only those in the `Apllications`?

> You're right, great catch. Anyway, that would probably influence only the Dropbox App definition, not the integration code. One would rather need to specify a certain directory outside of the Applications/dvc. I'll test that.

thanks! 🙏 
it is also about the DVC app and the app type we recommend creating.

> Honestly, I didn't even know one could do dvc -d dropbox://something -o dropbox://model.pkl. What changes would be required to make it work?

not need to do anything about it in this PR (may be some stubs with NotImplemented exception, we'll see)- we can address it later . It's more about me trying to understand better what type DVC Dropbox app should have. But it looks like we agreed that it should be the Scoped one, not the `App folder`.

Reply:
> Sorry, I haven't tried it. So, does it mean that if I have an app installed and have a pro account it won't be syncing any new directories, or only those in the Apllications?

You don't actually need a Pro account, that's one of the recent free features if I recall properly. More [here](https://www.dropbox.com/smart-sync).

> it is also about the DVC app and the app type we recommend creating.

I've just re-created my `dvc-test` app as scoped app with access to the whole Dropbox account instead of the App Folder. Works like a charm, without the limitation to the specific directory - you can setup any directory you wish, including a shared one. The permissions used are exactly the same as for the App Directory setup.

[Here](https://www.dropbox.com/lp/developers/reference/oauth-guide#implicit-grant) in the section "Using User and Team Tokens" it is stated though that enabling the Team scope would require a different approach. I couldn't make the app working with any Team scope enabled, as my subscription doesn't cover that. Probably one would need to define two separate apps (one for personal and one for Team scope), but I can't check that simply.

Reply:
> You don't actually need a Pro account

you are right, you don't. But you still need a paid plan. Mine is free and that's why I don't have smart sync. Not a problem at all- I would not expect DVC to be used a lot on free accounts.

> Works like a charm

perfect! That's how I initialize the `dvc` app already.

> Teams

I think it's good to take a look to at least understand the consequences and if we need to do something in advance. But overall it's fine to do support later. I'll read a bit more about apps and their differences. Primarily to do the right thing about the official DVC app.

I'll also enable the Teams subscription to test it, I'll invite you.